scipy/ndimage/tests/test_interpolation.py

```diff
@@ -898,6 +898,17 @@ def test_affine_transform_shift_reflect(self, order):
             x[::-1, ::-1],
         )
 
+    def test_affine_transform2(self):
+        x = numpy.zeros((100, 100, 100))
+        x[40:60, 40:60] = 1  # make rectangle
+        matrix = numpy.array([[1., 0., 0.],
+                              [0., 0.92, -0.38],
+                              [-0., 0.38, 0.92]])
+        shift = numpy.array([60, 60, 60])  # shift to corner of rectangle
+        origin = numpy.array([40, 40, 40])  # rotate around other corner
+        output = ndimage.affine_transform2(x, matrix, shift=shift, origin=origin)
+        assert numpy.abs(output.sum() - 31274.37) < 1
```

Comment:
You may see the CI failing about the `numpy` name because this branch gets merged with `main`, where we just started enforcing `import numpy as np` more stringently.

So, this diff will clean up the test in that regard:

```diff
--- a/scipy/ndimage/tests/test_interpolation.py
+++ b/scipy/ndimage/tests/test_interpolation.py
@@ -899,15 +899,15 @@ class TestNdimageInterpolation:
         )
 
     def test_affine_transform2(self):
-        x = numpy.zeros((100, 100, 100))
+        x = np.zeros((100, 100, 100))
         x[40:60, 40:60] = 1  # make rectangle
-        matrix = numpy.array([[1., 0., 0.],
+        matrix = np.array([[1., 0., 0.],
                               [0., 0.92, -0.38],
                               [-0., 0.38, 0.92]])
-        shift = numpy.array([60, 60, 60])  # shift to corner of rectangle
-        origin = numpy.array([40, 40, 40])  # rotate around other corner
+        shift = np.array([60, 60, 60])  # shift to corner of rectangle
+        origin = np.array([40, 40, 40])  # rotate around other corner
         output = ndimage.affine_transform2(x, matrix, shift=shift, origin=origin)
-        assert numpy.abs(output.sum() - 31274.37) < 1
+        assert np.abs(output.sum() - 31274.37) < 1
```

But, then you may see another test failure unless you also add the new function to `__all__`:

```diff
--- a/scipy/ndimage/_interpolation.py
+++ b/scipy/ndimage/_interpolation.py
@@ -41,7 +41,8 @@ from ._ni_docstrings import docfiller
 
 
 __all__ = ['spline_filter1d', 'spline_filter', 'geometric_transform',
-           'map_coordinates', 'affine_transform', 'shift', 'zoom', 'rotate']
+           'map_coordinates', 'affine_transform', 'shift', 'zoom', 'rotate',
+           'affine_transform2']
```

and then also switch to `np` in your new function.

Anyway, pretty minor drive-by comments, I guess we'll see what the `ndimage` regular have to say about the design.